java/flight/flight-jdbc-driver/src/main/java/org/apache/arrow/driver/jdbc/FlightClientCloser.java

```diff
@@ -0,0 +1,108 @@
+package org.apache.arrow.driver.jdbc;
+
+import org.apache.arrow.driver.jdbc.utils.FlightStreamQueue;
+import org.apache.arrow.flight.ArrowMessage;
+import io.grpc.stub.StreamObserver;
+import org.apache.arrow.flight.FlightDescriptor;
+import org.apache.arrow.flight.FlightStream;
+import org.apache.arrow.flight.sql.FlightSqlClient;
+import org.apache.arrow.memory.ArrowBuf;
+import org.apache.arrow.vector.VectorSchemaRoot;
+import org.apache.arrow.vector.dictionary.DictionaryProvider;
+import org.apache.arrow.vector.types.pojo.Schema;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.concurrent.CompletableFuture;
+
+public class FlightClientCloser implements FlightStream {
```

Comment:
Ensures FlightSqlClient is closed when the Stream is closed. I'm not sure if this is the right way to do it, but it seemed the least disruptive to the lifecycle of the driver and Future<Stream> stuff.